File: codegen/results/v001/sonnet/20260407_174952/python/004_json_filter/solution.py
import json
import sys

with open("input/data.json") as f:
    data = json.load(f)

filtered = [r for r in data if r.get("active") is True and r.get("age", 0) >= 30]
filtered.sort(key=lambda r: r.get("name", ""))

print(json.dumps(filtered))

File: codegen/results/v001/sonnet/20260407_174952/typescript/004_json_filter/solution.ts
import fs from "fs";
import path from "path";

interface Record {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

const data: Record[] = JSON.parse(fs.readFileSync(path.join("input", "data.json"), "utf-8"));

const result = data
  .filter((r) => r.active && r.age >= 30)
  .sort((a, b) => a.name.localeCompare(b.name));

console.log(JSON.stringify(result));
